### PR TITLE
Work in Progress: allow to assert on single-child elements

### PR DESCRIPTION
**Pull request by c089**
The README says:

> Assertions that test none, singular an plural of a possible collection generally work in all the forms ou would expect. For example to have no children, to have child, to have children.

but (besides being a bit hard to parse ;)) I couldn't figure out how to assert for things matching single results... I think the assertions are missing and added on example. Some other things could be:

```
expect(e, 'to have a single child matching', selector);
expect(e, 'when queried for only', selector);
// TODO oops, gotta run to a catch a train but I think there could be more things missing unless I just don't see an easy way to compose with an array expectation or something
```

### CODE REVIEW

**Comment by c089** [comment]
Also, I'm not quite sure on the best language to use, not being a native speaker. See e.g. [this question on the English stackExchange](https://english.stackexchange.com/questions/283671/is-there-a-word-for-an-only-child)

**Comment by sunesimonsen** [comment]
Sorry that nobody never got back to you, I guess we have lots you as a user by now. But wouldn't this not be more easily achieved this way:

```js
expect(e, 'queried for', selector, 'to have length', 1)
```

**Comment by c089** [comment]
Don't be sorry, I'm not entitled to your work and could use my fork if I needed to. I just didn't care enough to follow up. Checking on length works, just reads a bit less nice and just from reading the docs I felt it should work more like this. Feel free to close this though as I don't have a stake in this right now and probably won't prioritize working on this PR again soon. Happy new year :)

**Comment by sunesimonsen** [comment]
@c089 thanks - I'll leave it open for @Munter to consider. We can take the PR the rest of the way if anything need to be done, so no worries about having to follow up. 

@papandreou what do you think?

**Comment by papandreou** [comment]
Hmm, my gut feeling says that `to have a single child` isn't generally useful enough to be in the core lib. I could of course be wrong -- what is it used for?

@sunesimonsen, `expect(e, 'queried for', selector, 'to have length', 1)` isn't exactly the same as what's implemented here. That would only count the child nodes that are elements, where the `to have a single child` looks at `childNodes`, which could also contain text nodes, comment nodes etc. Not sure that's intentional, though?

**Comment by c089** [comment]
I believe my title was misleading. IIRC the intention here was never to use this matcher alone, but in combination with a query matcher as in "to have only one thing that matches x". But it's been too long, I can't remember the context and my test cases aren't helping at revealing my intention at all ;) Let's forget about it, at least until the day I will suddenly remember that use case I had last summer :D

**Comment by sunesimonsen** [comment]
@c089 I'll close the PR then :-)
